user-interface/6001-SORD-submit_orders.md

```diff
@@ -0,0 +1,139 @@
+# Submit order
```

Comment:
I assume this is a Console specific requirement, but similar ones will apply to other UIs. It's also not clear in some of the above files which fornt end / product the requirements apply to.

I think these files should be structured in folders or with names that make clear the project/product they apply to.

Reply:
Hmmm... I'm not sure I follow the purpose of these anymore tbh.

> Ideally I'd keep these agnostic of any implementation. they are more like things that users need than things that "console" should do.

Which users?
Is your plan to build multiple versions of these by user type/persona and user journey/use case?

> This way two different apps could choose how many or which of the should's they'd like to support.

How do they choose? 
What if noone _wants_ to support something that is a priority?
How do we prevent requirements relating to a specific user journey or use case being split across apps? 
How are the app devs/projects expected to engage with these?

> Similar-ish to how the spec describe the protocol not a particular node/client/package implementation.

The specs describe a **single** protocol and I'd sort of expect various sets of requirements to similarly each describe a single project/application, or at the most granular level, user journeys (but even then you may cover similar journeys in multiple apps for different types of user…)

> If this proves just too lofty an ideal we could refactor them to be app specific but then maybe they should not live in the specs repo?

I'd suggest maybe a discussion in cross-platform UX about what to do with these and how to tackle this higher level stuff.

It would be good to have a record of the requirements for Console to enable testing, planning, traceability from objective setting, etc. but now might not be the time for that. A quick refactor to make these app specific is tempting. Needs some thought…

Reply:
The other thing I would say is that many of these, although you say they are generic, are quite clearly not universal as it stands. I'd imagine for example that simple UI are actively not doing some of the must-haves here because they are targeting different users and user cases.

Reply:
> How do they choose?
What if noone wants to support something that is a priority?
How do we prevent requirements relating to a specific user journey or use case being split across apps?
How are the app devs/projects expected to engage with these?

For each app, there will be a list of ACs that a met/not met. It's up to that app to decide what to do with that. For the apps being built in vegaprotocol GitHub now I'd say there is a conversation to have about what of these we do want to meet for any given milestone.

Reply:
I think the thing that seems strange to me is where does "must" come from (for example) and what does it mean, if no dapp project has agreed they want to do that must and it hasn't come from our OKRs or prioritisation session? 

Seems there's now just a slightly odd extra way to add in requirements and how we come up with them and capture them and why they are there isn't clear, if they are not actually agreed to and claimed by a project already.

I guess I assumed initially that these ACs would be about things we had already agreed and decided were in scope in the normal way and had decided how they were delivered, so they would help us know what to test to prove we had done what we committed to. But it now also seems like a way that you (+ front end team???) are communicating some additional view of what we should do or what people should be building.

I don't necessarily think this is the wrong way to do that, and perhaps it traces back to existing OKRs or provides ACs that cover only agreed scope but that just doesn't seem clear to me at the moment, and if we are deciding new scope this way it probably needs more reviews and more input from other stakeholders.

In any case if everyone is comfortable with the scope of these ACs and how we plan on using them now I think we can refine what they are and what they're for later. I'm not too concerned by any of this at the moment and happy to punt some of the decisions and process questions down the road a bit. Having them is definitely better than not, even if we have a few things to figure out still.